### DISCUSSION TITLE
Apply new field filters to products list query

### DISCUSSION THREAD

**Discussion by BluetyIt** [swellstores/community#399] [Help] [answered]
Hello Guys,

We tried to list some products with custom filters. I tried to look directly in the documentation but I don't see any detail regarding how to add new **attributes filters**

swellClient.products.list({
      category: "",
      limit: DISPLAYED_PRODUCTS_NUMBER,
      page: pageParam, // Pass the page number to the API
      expand: ["brand_detail"],
      $filters: {
       "content.custom_field": "test"
        price: [100, 200],
      },
    });

I tried to "PUT" new permissions fields as recommended in documentation but it seems not working with filters or search function. Only to return additional fields in JSON.
https://developers.swell.is/guides/opening-content-permissions#using-input-fields

Thanks for your help

**Comment by logeshswell** [comment, marked as answer]
I assume that you use swell-js following this doc:
https://developers.swell.is/frontend-api/products

$filters won't work for custom fields. Instead you can try the below snippet:
```js
swellClient.products.list({
  ...
  expand: ["brand_detail"],
  content: {
    custom_field: "test"
  }
});
```
On a side note, you don't need to open read permissions for content fields which is opened by default. But, you need to open up input fields for write permissions following the same documentation.
